src/tcgwars/logic/impl/gen3/UnseenForces.groovy

```diff
@@ -421,16 +418,18 @@ public enum UnseenForces implements LogicCardInfo {
           energyCost C, C
           onAttack {
             damage 20
-
-            delayed {
-              before BETWEEN_TURNS, {
-                if (bg.currentTurn == self.owner.opposite) {
-                  directDamage 30, self.owner.opposite.pbg.active
-                  bc "Spiky Shell activates"
+            targeted(defending){
+              delayed {
+                before BETWEEN_TURNS, {
+                  if (bg.currentTurn == self.owner.opposite) {
+                    bc "Spiky Shell activates"
+                    directDamage 30, defending
+                  }
                 }
+                unregisterAfter 2
+                after SWITCH, defending, {unregister()}
+                after EVOLVE, defending, {unregister()}
               }
```

Comment:
`targeted` wrap is not needed for effects that already target internally.  in this case `directDamage 30, defending` will target correctly.

Reply:
Edited it based on a similar interaction that was patched recently (Trevenant&Dusknoir-GX). Right now on Dev, a Pokémon preventing all effects of attack will still be hit by the delayed directDamage. the targeted should prevent the delayed effect being placed in the first place.